### PR TITLE
[RW-4709][risk=no] Add jest-junit reporter in e2e project

### PR DESCRIPTION
**Pull request by aweng98**

### CODE REVIEW

File: e2e/package.json [hunk 1]
```diff
@@ -39,7 +43,7 @@
     "faker": "^4.1.0",
     "fs-extra": "^9.0.0",
     "jest": "^25.1.0",
-    "jest-cli": "^25.1.0",
```
Comment (aweng98) [inline e2e/package.json:42]: I am removing `jest-cli` for now because I haven't used any of its feature.

File: .circleci/config.yml [hunk 2]
```diff
@@ -298,15 +298,15 @@ jobs:
           command: git submodule update --init --recursive
       - restore_cache:
           keys:
-          - e2e-test-cache-{{ checksum "~/workbench/e2e/yarn.lock" }}
-          - e2e-test-cache-
+          - v1-e2e-test-cache-{{ checksum "~/workbench/e2e/yarn.lock" }}
```
Comment (aweng98) [inline .circleci/config.yml:301]: Adding a version number. From [CircleCI official doc](https://circleci.com/docs/2.0/caching/):
`If you would like to invalidate your cache manually, you can bump v1 to v2 in your config.yml file. In this case, you would now have a new cache key v2-npm-deps, which will trigger the storing of a new cache.`

File: .circleci/config.yml [hunk 3]
```diff
@@ -317,9 +317,12 @@ jobs:
       - run:
           name: Run e2e tests (Puppeteer and Jest)
           working_directory: ~/workbench/e2e
-          command: yarn --silent test:ci
+          command: yarn test:ci
       - store_artifacts:
           path: e2e/logs
+          destination: logs
+      - store_test_results:
+          path: ~/workbench/e2e/logs
```
Comment (blrubenstein) [inline .circleci/config.yml:325]: Just to check, this is writing to circle, right? and nowhere else?

Comment (aweng98) [reply .circleci/config.yml:325]: In `e2e` project, any screenshot or/and html file are saved in `/logs` sub-directory. If you run tests on localhost, you'll find failed screenshots and html files in `/logs` dir. 
We have to set `store_test_results` key to tell CircleCI where to find saved files.

**Pull request merged**